opentelemetry-sdk/src/opentelemetry/sdk/_logs/_internal/__init__.py

```diff
@@ -662,16 +666,28 @@ def get_logger(
                 schema_url=schema_url,
                 attributes=attributes,
             )
-        return Logger(
-            self._resource,
-            self._multi_log_record_processor,
-            InstrumentationScope(
-                name,
-                version,
-                schema_url,
-                attributes,
-            ),
-        )
+        key = (name, version, schema_url)
```

Comment:
The specification says that the Instrumentation scope is identified by (name, version) pair https://github.com/open-telemetry/opentelemetry-specification/blob/7284c123c9889f88ebc31d288d2d108b5ff9e0d9/specification/glossary.md#instrumentation-scope

Reply:
I think makes sense. Also, we recently introduced the support to optional scope attributes for logs here https://github.com/open-telemetry/opentelemetry-python/pull/4035

Reply:
To comply with this, I made the change not cache if attributes are specified and cache otherwise. Do you think this would be an acceptable approach?

One workaround to cache attributes would be to do `json.dumps(attributes, sort_keys=True)`, however it is slow and probably not a good practice.

Reply:
I agree it would probably be too slow. Maybe this is all moot since it's not currently possible to use the scope attributes with stdlib logging handler anyway.

If someone is using the Events API and passing scope attributes, they can just manage a reference themselves.